File: C/tree.c
#include <stdio.h>
#include "tree.h"
#include "treeNode.h"

TREE constructorTree() {
	TREE tree;
	tree.root = NULL;

	return tree;
}

void addValueInTree(PTREE tree, int value) {
   	PNODE newNode = constructor(value);
   	tree->root = addNode(tree->root, newNode);
}

void removeValueInTree(PTREE tree, int value) {
   	tree->root = removeNode(tree->root, value);
}

int numberOfNodesInTree(PTREE tree) {
   	return countNodes(tree->root);
}

void searchValueInTree(PTREE tree, int value) {
	PNODE result = searchNode(tree->root, value);

	if (result == NULL) printf("VALOR %d NAO ENCONTRADO\n", value);
	else printf("VALOR %d ENCONTRADO\n", result->valor);
}

void printTree(PTREE tree) {
	printf("ARVORE: ");
	printNodes(tree->root);
	printf("\n");
}

File: JS/tree.js
import TreeNode from './treeNode.js'

class Tree {
  #root

  constructor () {
    this.#root = null
  }

  addValue (value) {
    var newNode = new TreeNode(value)
    this.#root = TreeNode.addNewNode(this.#root, newNode)
  }

  searchValue(value) {
    var result = TreeNode.searchValue(this.#root, value)

    if (result) console.log('VALOR ENCONTRADO:', result.value)
    else console.log('VALOR NÃO ENCONTRADO')
  }

  deleteValue(value) {
    this.#root = TreeNode.deleteData(this.#root, value)
  }

  print() {
    console.log('ÁRVORE')
    TreeNode.print2DTree(this.#root)
    console.log('')
  }

  numberOfNodesInTree() {
    return TreeNode.countNodes(this.#root)
  }
}

export default Tree
